eogrow/core/eopatch.py

```diff
@@ -253,6 +249,19 @@ def generate_names(self, bbox_dataframe: GeoDataFrame, *_: Any, **__: Any) -> bi
 class BatchTileManager(EOPatchManager):
     """A custom patch manager that uses a naming convention based on Sentinel Hub Batch tiles"""
 
-    def generate_names(self, bbox_dataframe: GeoDataFrame, *_: Any, **__: Any) -> bidict:
+    _area_manager: BatchAreaManager
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+
+        if not isinstance(self._area_manager, BatchAreaManager):
+            raise ValueError(f"{self.__class__.__name__} is only compatible with {BatchAreaManager.__name__}")
```

Comment:
it might be better to update the `Schema` of the manager, so that this is caught at validation time. We do need to test that `pydantic` actually does what it promises (so that it will check that the manager returned by the validator is of the correct type). Keep the `_area_manager: BatchAreaManager` though, it is better than casts as it checks for subtypes correctly

Reply:
This whole fiasco works because under the hood `_area_manager` is of type `Any` (thanks to the `load_manager` function). I am considering adding such a safety net there, somehow (so automatically checking that the annotation and the manager match). Maybe I should open an issue?

Reply:
Note that schemas for area manager and EOPatch manager are 2 different subparts of pipeline config. How would you validate that 1 schema is compatible with another? (I wouldn't put this check into the base `Pipeline.Schema` validation because it's way too specific.)

Reply:
ah, nevermind, I got confused on the schema part

Here I would annotate `area_manager` and not `_area_manager`, so that the *type correction* is present in the __init__as well

Reply:
Ah, I see it does not get propagated properly.